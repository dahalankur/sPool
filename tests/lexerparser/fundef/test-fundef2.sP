# a recursive function that uses automatic memoization
def store int fibonacci(int n):
    if (n <= 1):
        return n
    else
        return fibonacci(n-1) + fibonacci(n-2); # this semi-colon closes the if
    ; # closes the def