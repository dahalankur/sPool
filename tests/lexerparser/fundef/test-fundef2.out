PROGRAM {[ 
    FUN(STORE, INTTY, fibonacci, FORMALS((INTTY, n)), [ 
        IF(BINOP(VAR(n), LEQ, INT(1)), [ 
            RETURN(VAR(n))], [ 
            RETURN(BINOP(CALL(fibonacci, ARGS(BINOP(VAR(n), MINUS, INT(1)))), PLUS, CALL(fibonacci, ARGS(BINOP(VAR(n), MINUS, INT(2))))))])])]}
